.github/workflows/ci_workflows.yml

```diff
@@ -9,11 +9,11 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Checkout code
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
       with:
         fetch-depth: 0
     - name: Set up Python
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
         python-version: '3.x'
```

Comment:
somewhat related: with `setup-python@v4` you can also use `cache: 'pip'` and `cache-dependency-path: 'setup.cfg'` to cache the Python install based on the job name, until the dependencies change. This can speed up subsequent runs

Reply:
The CI here is already very fast. But if you want to add caching, feel free in a follow-up PR.